bind.go

```diff
@@ -189,6 +213,86 @@ func (ml *maybeLineage) checkLensesOrder() error {
 	return nil
 }
 
+func (ml *maybeLineage) checkGoLensCompleteness() error {
+	all := make(map[lensID]bool)
+	for _, lens := range ml.implens {
+		all[lid(lens.To, lens.From)] = true
+	}
+
+	var missing []lensID
+
+	var prior SyntacticVersion
+	for _, sch := range ml.schlist[1:] {
+		// there must always at least be a reverse lens
+		v := sch.Version()
+		revid := lid(prior, v)
+
+		if !all[revid] {
+			missing = append(missing, revid)
+		} else {
+			delete(all, revid)
+		}
+
+		if v[0] != prior[0] {
+			// if we crossed a major version, there must also be a forward lens
+			fwdid := lid(v, prior)
+			if !all[fwdid] {
+				missing = append(missing, fwdid)
+			} else {
+				delete(all, fwdid)
+			}
+		}
+		prior = v
+	}
+
+	// TODO is it worth making each sub-item into its own error type?
```

Comment:
Personally, I don't see any need (at least for now), to explicitly model every item into its own error, as far as we bring that information back to the consumer.